public/request/game/modify.php

```diff
@@ -1,48 +1,53 @@
 <?php
 
 use RA\ArticleType;
+use RA\GameAction;
 use RA\Permissions;
 
 require_once __DIR__ . '/../../../vendor/autoload.php';
 require_once __DIR__ . '/../../../lib/bootstrap.php';
 
-if (!ValidatePOSTChars("ugfv")) {
+if (!ValidatePOSTChars("gfv")) {
     echo "FAILED";
     exit;
 }
 
-$author = requestInputPost('u');
-$gameID = requestInputPost('g');
-$field = requestInputPost('f');
+$gameID = (int) requestInputPost('g');
+$field = (int) requestInputPost('f');
 $value = requestInputPost('v');
 
-if (authenticateFromCookie($user, $permissions, $userDetails, Permissions::Developer)) {
-    if ($field == 4) {
-        $name = requestInputPost('n');
-        $labels = requestInputPost('l');
-        if (updateHashDetails($gameID, $value, $name, $labels)) {
-            // Log hash update
-            addArticleComment("Server", ArticleType::GameHash, $gameID, $value . " updated by " . $user . ". Description: \"" . $name . "\". Label: \"" . $labels . "\"");
-            echo "OK";
-        } else {
-            echo "FAILED!";
-        }
+if (!authenticateFromCookie($user, $permissions, $userDetails, Permissions::Developer)) {
+    echo "FAILED!";
+    exit;
+}
+
+if ($field === GameAction::UpdateHash) {
+    $name = requestInputPost('n');
+    $labels = requestInputPost('l');
+    if (updateHashDetails($gameID, $value, $name, $labels)) {
+        // Log hash update
+        addArticleComment("Server", ArticleType::GameHash, $gameID, $value . " updated by " . $user . ". Description: \"" . $name . "\". Label: \"" . $labels . "\"");
+        echo "OK";
         exit;
     }
+    echo "FAILED!";
+    exit;
+}
 
-    if (modifyGame($author, $gameID, $field, $value)) {
-        if ($field == 3) { // Only return status when unlinking hash
-            echo "OK";
-            exit;
-        }
-        header("location: " . getenv('APP_URL') . "/game/$gameID?e=modify_game_ok");
+if (modifyGame($user, $gameID, $field, $value)) {
+    if ($field == GameAction::UnlinkHash) {
+        // Only return status when unlinking hash
+        echo "OK";
         exit;
-    } else {
-        if ($field == 3) { // Only return status when unlinking hash
-            echo "FAILED!";
-            exit;
-        }
-        header("location: " . getenv('APP_URL') . "/game/$gameID?e=errors_in_modify_game");
+    }
+    header("location: " . getenv('APP_URL') . "/game/$gameID?e=modify_game_ok");
```

Comment:
Didn't find enough time to untangle this but this request script should _either_ response to ajax requests (with structured json data if possible, not just echo'd strings) _or_ form requests that issue redirects. I'd prefer the latter.
An attempt for simplification will probably follow in a later PR.

Reply:
I'm going to be touching this as part of the game audit comments feature. What exactly are you looking for here?

Reply:
https://github.com/RetroAchievements/RAWeb/pull/1017/files#diff-52e0986859da92ea38c57b01ddcee4160047d764052aeb975ffe7b3cc62f0632

Basic forms and location header redirects if possible (_If_ it makes sense to implement it with javascript XHR requests then it should at least respond with JSON. we really shouldn't be using javascript for form requests though imo)